features/rspec_runner_spec.rb

```diff
@@ -19,6 +21,36 @@
 
     is_expected.to match(/Prediction size \d+ is over the limit \(1\)/)
       .and match(/Prediction is pruned to fit the limit!/)
-      .and match(/1 example, 0 failures/)
+      .and match(/\d+ examples?, 0 failures/)
+  end
+
+  context 'when file, spec id, and directory are predicted' do
+    before do
+      ENV['CRYSTALBALL_EXAMPLES_LIMIT'] = '0'
+      ENV['CRYSTALBALL_PREDICTION_BUILDER_CLASS_NAME'] = 'PredictionBuilder'
+      ENV['CRYSTALBALL_REQUIRES'] = './prediction_builder'
+    end
+
+    after do
+      ENV.delete('CRYSTALBALL_EXAMPLES_LIMIT')
+      ENV.delete('CRYSTALBALL_PREDICTION_BUILDER_CLASS_NAME')
+      ENV.delete('CRYSTALBALL_REQUIRES')
+    end
+
+    it 'runs the whole file' do
+      change other_important_class_path # Adds ./spec/class2_spec.rb
+      change class1_path                # Adds ./spec/class2_spec.rb[1:1:1]
+
+      is_expected.to match(/.another_field/) # ./spec/class2_spec.rb[1:2]
+    end
+
+    context 'when the files are contained in the directories' do
+      it 'runs the whole directory' do
+        change important_class_path            # Adds ./spec/important_dir/
+        change class2_path, 'class Class2;end' # Adds ./spec/important_dir/important_spec.rb[1:2]
+
+        is_expected.to match(/does very specific stuff/) # ./spec/important_dir/important_spec.rb[1:1]
+      end
+    end
```

Comment:
where is the check about edge case you missed before? `class1_spec.rb[1:1] class2_spec.rb[1:2]` should NOT run full `class1_spec.rb`?